### PR TITLE
Calendar can now be populated with arbitrary components/added event due date

### PR DESCRIPTION
**Pull request by felixangell**
The calendar API was changed so that arbitrary components can be added to a cell.

Events now show their due date even from the click of a button.
In addition, you now press a button to inspect a GLP rather than clicking the events text.

If the ghost event is on the same calendar month the calendar refreshes and it is rendered. Otherwise, the calendar will change to the month where the due date occurs. 

When clicking a due date event, it is removed from the calendar view. (can be removed/changed)

Only one due date is shown at a time (this can be changed in the future if necessary)

### CODE REVIEW

**Comment by juddus** [comment]
:+1: 

For objects with the same key as var name, you don't need to put the name twice:
```js
{
    id: id,
}
```
can also be
```js
{
    id,
}
```

File: frontend/src/modules/calendar/root/calendar/calendar_event.js [hunk 1]
```diff
@@ -1,19 +1,59 @@
...
+                            id: id,
+                            name: name,
+
+                            // for now we render the due date as today
+                            due: due,
+                        });
+                    },
+                }, span({
+                    'title': 'Go to due date',
+                }, i('.icon-clock')))
```
Comment (felixangell) [inline frontend/src/modules/calendar/root/calendar/calendar_event.js]: @juddus Is this ok? Is there a cleaner way to do this other than with a ternary and an empty array?

Comment (juddus) [reply frontend/src/modules/calendar/root/calendar/calendar_event.js]: You could separate the logic into a do statement if you wanted.

```js
const dueEventEl = do {
    if (nullishCheck(due, false)) {
        a(
            '.fake-link',
            {
                target: '_blank',
                onclick: () => {
                    this.emit('WriteDueEvent', {
                        id: id,
                        name: name,

                        // for now we render the due date as today
                        due: due,
                    });
                },
            }, 
            span(
                {
                    'title': 'Go to due date',
                },
                i('.icon-clock')
            )
        );
    } else {
        [];
    }
};

return div(
    '.event',
    p('.event-name', name),
    div(
        a(
            {
                href: `//${window.location.host}/lesson_manager/#view?id=${id}`,
                target: '_blank',
            }, 
            span(
                {
                    'title': 'Inspect GLP',
                },
                i('.icon-search'),
            )
        ),
        dueEventEl,
    ),
);
```
Also be careful with the readability of the elements, sometimes it's hard to know where the element is in the dom tree.

Comment (felixangell) [reply frontend/src/modules/calendar/root/calendar/calendar_event.js]: Hmmm yeah that's a bit nicer, cheers

**Pull request merged**